integrations/bing-ads/lib/index.js

```diff
@@ -82,5 +89,35 @@ Bing.prototype.track = function(track) {
   if (track.category()) event.ec = track.category();
   if (track.revenue()) event.gv = track.revenue();
 
+  if (this.options.enableConsent) {
+    // eslint-disable-next-line no-use-before-define
+    updateConsent.call(this, track);
+  }
+
   window.uetq.push(event);
 };
+
+function updateConsent(track) {
+  var consent = {};
+
+  var propertiesPath = track.proxy(
```

Comment:
nit: It is generally not a good idea to mutate properties as it is hard to read and debug. A better approach will be to set consent based on consent category and return early. And then parse propertiesPath